lib/carrierwave/downloader/base.rb

```diff
@@ -18,9 +18,11 @@ def initialize(uploader)
       # === Parameters
       #
       # [url (String)] The URL where the remote file is stored
+      # [download_retry_count （Int）] Retry count when download failed
       # [remote_headers (Hash)] Request headers
       #
-      def download(url, remote_headers = {})
+      def download(url, download_retry_count = 0, remote_headers = {})
```

Comment:
Please don't insert `download_retry_count` into arguments, this will break existing implementations.

A downloader instance have a reference to an uploader instance, you can just reference `uploader.download_retry_count` inside the downloader.